build_tools/testing/run_jax_tests.sh

```diff
@@ -44,7 +44,18 @@ diff_jax_test() {
     echo "no difference found"
 }
 
-diff_jax_test test/test_simple.py
+# FIXME: due to #19223, we need to use jax no higher than 0.4.20,
+# but in such version of jax, 'stablehlo.broadcast_in_dim' op
+# will be emitted without attribute 'broadcast_dimensions',
+# which leads to an error in IREE PJRT plugin.
+# So currently any program with broadcast will fail,
+# e.g. test/test_simple.py.
+# After #19223 is fixed, we can uncomment the line below.
+
+# diff_jax_test test/test_simple.py
+
+diff_jax_test test/test_add.py
```

Comment:
As the FIXME here, in JAX 0.4.20,  'stablehlo.broadcast_in_dim' op will be emitted without attribute 'broadcast_dimensions', which will make the IREE fail to compile. So currently some existing tests like `integerations/pjrt/test/test_simple.py` cannot pass.

I add a new (and simpler) test `test_add.py` instead of running `test_simple.py`. And I observed that after #19223 is fixed (e.g via the patch #19241), and JAX is updated to the latest version, this error will disappear.

Reply:
Makes sense. Good to know that rolling the versions forward will enable more tests to pass.

StableHLO does have a mechanism to actually be "stable" within a support window - using VHLO ("Versioned StableHLO"):

* https://github.com/openxla/stablehlo/blob/main/docs/compatibility.md
* https://github.com/openxla/stablehlo/blob/main/docs/vhlo.md

It's been a while since I've looked at the specifics across projects, but if JAX can emit VHLO instead of StableHLO then usage like this should be more robust. Since https://github.com/iree-org/iree/commit/4b9755f63e13bb81e4a1e14c0cfddccb69a8d58c, IREE's import pipeline has a "deserialize VHLO to StableHLO" pass that runs close to the start.